src/Cake.Core/Modules/CoreModule.cs

```diff
@@ -36,6 +36,7 @@ public void Register(ICakeContainerRegistrar registrar)
             registrar.RegisterType<DefaultExecutionStrategy>().As<IExecutionStrategy>().Singleton();
 
             // Environment
+            registrar.RegisterType<LaunchInfo>().As<ILaunchInfo>().Singleton();
```

Comment:
I could be wrong, but I don't think you need to register `LaunchInfo` again here - I'd expect the same instance to flow through to this scope. Could you take a look?

If it's really needed, then I think we need to find a way to get the instance created at the start and pass it through here, so you can register the same instance, because creating a new instance of `LaunchInfo` here might be too late because bootstrapping could already have happened and changed the working directory.